libs/net/nodenetworkconfigurationpolicy.py

```diff
@@ -0,0 +1,183 @@
+
+from dataclasses import asdict, dataclass
+from typing import Any
+
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.exceptions import NNCPConfigurationFailed
+from ocp_resources.resource import Resource, ResourceEditor
+from timeout_sampler import retry
+
+from utilities.constants import TIMEOUT_5SEC, TIMEOUT_90SEC
+
+
+@dataclass
+class STP:
+    enabled: bool
+
+
+@dataclass
+class BridgeOptions:
+    stp: STP
+
+
+@dataclass
+class Port:
+    name: str
+
+
+@dataclass
+class Bridge:
+    options: BridgeOptions | None = None
+    port: list[Port] | None = None
+    allow_extra_patch_ports: bool | None = None
+
+
+@dataclass
+class IP:
+    auto_dns: bool = True
+    dhcp: bool = False
+    enabled: bool = False
+
+
+@dataclass
+class IPv4(IP):
+    pass
+
+
+@dataclass
+class IPv6(IP):
+    autoconf: bool = False
+
+
+@dataclass
+class Interfaces:
+    name: str
+    type: str
+    state: str
+    ipv4: IPv4
+    ipv6: IPv6
+    bridge: Bridge | None = None
+
+
+@dataclass
+class DesiredState:
+    interfaces: list[Interfaces]
+
+
+class NodeNetworkConfigurationPolicy(Resource):
+    """
+    NodeNetworkConfigurationPolicy object.
+    """
+
+    api_group = Resource.ApiGroup.NMSTATE_IO
+
+    class Conditions:
+        class Type:
+            DEGRADED = "Degraded"
+            AVAILABLE = "Available"
+
+        class Reason:
+            CONFIGURATION_PROGRESSING = "ConfigurationProgressing"
+            SUCCESSFULLY_CONFIGURED = "SuccessfullyConfigured"
+            FAILED_TO_CONFIGURE = "FailedToConfigure"
+            NO_MATCHING_NODE = "NoMatchingNode"
+
+    def __init__(
+        self,
+        name: str,
+        desired_state: DesiredState,
+        client: DynamicClient | None = None,
+        node_selector: dict[str, str] | None = None,
+    ):
+        """
+        Create and manage NodeNetworkConfigurationPolicy
+
+        Args:
+            name (str): Name of the NodeNetworkConfigurationPolicy object.
+            desired_state (DesiredState): Desired policy configuration - interface creation, modification or removal.
+            client (DynamicClient, optional): Optional DynamicClient to use.
+            node_selector (dict, optional): A node selector that specifies the nodes to apply the node network
+                configuration policy to.
+        """
+        super().__init__(
+            name=name,
+            client=client,
+        )
+        self._desired_state = desired_state
+        self._node_selector = node_selector
+
+    @retry(
+        wait_timeout=TIMEOUT_90SEC,
+        sleep=TIMEOUT_5SEC,
+    )
+    def wait_for_status_success(self) -> list[dict]:
+        failed_condition_reason = self.Conditions.Reason.FAILED_TO_CONFIGURE
+        no_match_node_condition_reason = self.Conditions.Reason.NO_MATCHING_NODE
+        nncp_conditions = self.instance.status.conditions
+        for nncp_condition in nncp_conditions:
+            if nncp_condition["status"] == f"{True}":
```

Comment:
Why not using:
```suggestion
            if nncp_condition["status"] == self.Condition.Status.TRUE:
```

To be more consistent like you used here:
https://github.com/RedHatQE/openshift-virtualization-tests/blob/8e3c3e4e07e7e2449cfadb3d0c6239fa56497edc/libs/net/nodenetworkconfigurationpolicy.py#L122